airbyte-cdk/python/file_based_build_customization.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from __future__ import annotations
+
+import textwrap
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from dagger import Container
+
+
+def setup_nltk(connector_container: Container) -> Container:
+    """
+    Seeds the connector with nltk data at build time. This is because the nltk data
+    is large and takes a long time to download. It runs a python script that downloads
+    the data following connector installation.
+    """
+
+    nltk_python_script = textwrap.dedent(
+        """
+        import nltk
+        nltk.download('punkt')
+        nltk.download('averaged_perceptron_tagger')
+        """
+    )
+    connector_container = (
+        connector_container.with_new_file("/tmp/nltk_python_script.py", nltk_python_script)
+        .with_exec(["python", "/tmp/nltk_python_script.py"], skip_entrypoint=True)
+        .with_exec(["rm", "/tmp/nltk_python_script.py"], skip_entrypoint=True)
+    )
+
+    return connector_container
+
+
+def install_tesseract_and_poppler(connector_container: Container) -> Container:
+    """
+    Installs Tesseract-OCR and Poppler-utils in the container. These tools are necessary for
+    OCR (Optical Character Recognition) processes and working with PDFs, respectively.
+    """
+
+    connector_container = connector_container.with_exec(
+        ["sh", "-c", "apt-get update && apt-get install -y tesseract-ocr poppler-utils"], skip_entrypoint=True
+    )
+
+    return connector_container
+
+
+async def post_connector_install(connector_container: Container) -> Container:
+    """
+    Handles post-installation setup for the connector by setting up nltk and
+    installing necessary system dependencies such as Tesseract-OCR and Poppler-utils.
+
+    These steps are necessary if the unstructured parser from the file based CDK is exposed in the connector.
+    """
+
+    # Setup nltk in the container
+    connector_container = setup_nltk(connector_container)
```

Comment:
Something interesting to note here, which is playing against the bundling of the nltk data in the base image, is that:
- this execution would depend of the installation of `nltk` in the  base image
- the result of this execution (the file download) is dynamic. We don't control what's being downloaded exactly

This is why it makes sense for it to be a `post_connector_install` thing.
If we downloaded this data in the base image we'd have to make sure the nltk version we download this data with is the same as the one declared in the CDK... 

I'm a bit concerned by the fact that this `post_connector_install` can't guarantee reproducible results as we rely on `nltk` to decide which data is downloaded.
@flash1293 could we host snapshots of this data on one of our public bucket and make this hook download this data and mount it to the container?

Reply:
Inlined the nltk data index in the customization script so it points to a raw github url of that specific commit - I have a pretty high confidence this is always reproduceable (short of the maintainers deleting the repo and then we have problems anyway)